libcxx/include/__new/allocate.h

```diff
@@ -47,52 +47,61 @@ _LIBCPP_HIDE_FROM_ABI void __libcpp_operator_delete(_Args... __args) _NOEXCEPT {
 #endif
 }
 
+template <class _Tp>
 inline _LIBCPP_HIDE_FROM_ABI void* __libcpp_allocate(size_t __size, size_t __align) {
```

Comment:
```suggestion
template <class _Tp>
inline _LIBCPP_HIDE_FROM_ABI _Tp* __libcpp_allocate(__element_count __size, size_t __align = _LIBCPP_ALIGNOF(_Tp)) {
  _LIBCPP_ASSERT_INTERNAL(__align >= _LIBCPP_ALIGNOF(_Tp), "Tried to allocate underaligned buffer");
```
Since we pass the type anyways we might as well improve the API.

Reply:
Right now, `__libcpp_allocate` takes a number of bytes. Are you suggesting we change that to a number of elements of type `_Tp`?

Reply:
Exactly.